make/aarch64_check_codec_order.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# **********************************************************
+# Copyright (c) 2018 Arm Limited    All rights reserved.
+# **********************************************************
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of VMware, Inc. nor the names of its contributors may be
+#   used to endorse or promote products derived from this software without
+#   specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+# Usage: ./aarch64_codec_codec_order.sh <path to core/arch/aarch64/> <path to build dir>
+
+# Check if operand patterns are ordered by pattern.
+cat $1/codec.txt | \
```

Comment:
Hmm, when we moved to CMake we wanted to avoid any dependencies on UNIX tools.  We got rid of all the uses of sed, awk, grep, cat, cut, head, tail, wc, bash, etc., replacing them with CMake.  That made everything cross-platform.  (We wanted to get rid of perl too but never finished that: #68).  So in that sense this seems like a step backward.  I know it is a pain to write file and regex utilities in cmake, and you probably don't want to do that.  Please at least include a comment at the top of this file acknowledging that this is not cross-platform, that this file should *not* be taken as a model for how to write new build utilities for DR, and list the assumptions it makes on all these utility apps being available.

Reply:
(Normally CMake would search for and validate utility apps, rather than a blind attempt to execute them.)